app/src/main/java/com/afwsamples/testdpc/common/RestrictionManagerCompat.java

```diff
@@ -31,6 +31,12 @@
 public class RestrictionManagerCompat {
```

Comment:
Modifying RestrictionManagerCompat to store some extra information for the UI sounds a bit hacky to me.
Also, RestrictionManagerCompat is ported from android.content.RestrictionsManager, so keeping it in sync with the framework make us easier to maintain.

IMO, the root cause of the bug is because we are mixing value(Bundle) and format(RestrictionEntry). They are actually two different things, but we keep converting between them!
How about having the fragment takes both value(Bundle) and format(RestrictionEntry) as input and render accordingly? With this, we can remove all the conversion code too.
It may requires a big change, so feel free to delegate to us if you do not have time. That said, we may not have it fixed in the near future as we are focusing on other things right now.

Feel free to chime in if you have any idea to get it fixed in a simple and clean way :)

Thanks!